lib/containers/engine.pm

```diff
@@ -321,6 +321,20 @@ sub cleanup_system_host {
     my ($self, $assert) = @_;
     $assert //= 1;
     $self->_engine_assert_script_run("ps -q | xargs -r " . $self->runtime . " stop", 180);
+
+    # all containers should be stopped before running prune
+    # https://github.com/containers/podman/issues/19038
+    my $count = 5;
+    my $no_container;
+    do {
+        $no_container = $self->_engine_script_output('ps --noheading');
+        $count--;
+
+        if ($no_container) {
+            record_soft_failure("bsc#1212825 - podman system prune sometimes fails or podman stop doesn't wait for containers to stop");
```

Comment:
```suggestion
            record_soft_failure("bsc#1212825 - podman system prune sometimes fails or podman stop doesn't wait for containers to stop");
            break;
```

_Gimme a break_. XD

Reply:
:D actually, the idea was to make sure all are stopped. So the loop would if the variable is empty. Anyway, just added `rm`